hw/ip/otbn/dv/otbnsim/sim/dmem.py

```diff
@@ -208,23 +150,20 @@ def changes(self) -> Sequence[Trace]:
     def _commit_store(self, item: TraceDmemStore) -> None:
         if item.is_wide:
             assert 0 <= item.value < (1 << 256)
-            self.data[item.addr // 32] = item.value
-            return
+            mask = (1 << 32) - 1
+            for i in range(256 // 32):
+                wr_data = (item.value >> (i * 32)) & mask
+                self.data[(item.addr // 4) + i] = wr_data
 
-        idx32 = item.addr // 4
-        idxW = idx32 // 8
-        offW = idx32 % 8
+            return
+        else:
+            self.data[item.addr // 4] = item.value
 
         # Since we store data in wide form, we have to do a read/modify/write
```

Comment:
I think this comment and everything else below can probably be removed from this function now

Reply:
Removed the comments